src/util/system.h

```diff
@@ -158,12 +158,18 @@ struct SectionInfo
 class ArgsManager
 {
 public:
+    /**
+     * Flags controlling how config and command line arguments are validated and
+     * interpreted.
+     */
     enum Flags : uint32_t {
-        // Boolean options can accept negation syntax -noOPTION or -noOPTION=1
-        ALLOW_BOOL = 0x01,
-        ALLOW_INT = 0x02,
-        ALLOW_STRING = 0x04,
-        ALLOW_ANY = ALLOW_BOOL | ALLOW_INT | ALLOW_STRING,
+        ALLOW_ANY = 0x01,    //!< disable validation
```

Comment:
I wonder whether `enum class Flags : uint32_t` may be used here as Jon did in https://github.com/bitcoin/bitcoin/pull/21506/ ([guideline](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Renum-class)). Would it make sense to you in this or a follow-up PR?

Reply:
I think the rationale in the guideline makes sense and that's why I suggested it. There was no specific advatange I had in mind. Anyway, it may (not) be a good follow up PR.

Reply:
See `NetPermissionsFlags` in net_permissions.h for comparison; it gives you lots of namespacing (so you'd be writing `ArgsManager::Flags::ALLOW_BOOL` everywhere) and means you can't just treat them as ints (so bitwise or'ing things together means you have to declare your own operator)

Reply:
Yes, two sides of a coin, I suppose.

Namespacing can allow shortening the enumerator names, since they no longer need to be unique in global namespace (and can therefore no longer need to be shouty "there can only be one"), e.g. ArgsManager::ALLOW_STRING could be ArgsManager::Flags::string. Whether that naming is better may be personal taste, but getting them out of global space seems good.

Using an enum class permits defining allowed operations: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Renum-oper and https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Renum-class ("minimize surprises: traditional enums convert to int too readily").

Reply:
(and changing NetPermissionFlags to an enum class found a bug for that reason.)